tests/lib/rules/no-useless-rest-spread.js

```diff
@@ -127,11 +115,27 @@ ruleTester.run("no-useless-rest-spread", rule, {
             output: "let obj = { ...b}",
             errors: ["Unexpected a spread property."],
         },
-        // Espree threw a syntax error.
-        // {
-        //     code: "let {a, ...{}} = obj",
-        //     output: "let {a} = obj",
-        //     errors: ["Unexpected a rest property."],
-        // },
+    ],
+})
+
+const babelRuleTester = new RuleTester({ parser: "babel-parser" })
+babelRuleTester.run("no-useless-rest-spread with babel", rule, {
+    valid: [],
+    invalid: [
+        {
+            code: "let {a, ...{b, c, ...d}} = obj",
+            output: "let {a, b, c, ...d} = obj",
+            errors: ["Unexpected a rest property."],
+        },
+        {
+            code: "let {a, ...{b, c, }} = obj",
+            output: "let {a, b, c} = obj",
+            errors: ["Unexpected a rest property."],
+        },
+        {
+            code: "let {a, ...{}} = obj",
```

Comment:
This is a syntax error in the final spec. `ObjectPattern` and `ArrayPattern` cannot be after rest operator.

- https://github.com/tc39/proposal-object-rest-spread/pull/50
- https://github.com/tc39/proposal-object-rest-spread/pull/51

Reply:
Probably worth just ditching the tests then?

Reply:
ESLint's test runner cannot include test cases which include syntax errors. Those are failed always even if it's in invalid tests. So I merely removed those commented-out codes in the master branch.

Reply:
@mysticatea cool, I closed the PR